packages/platform-core/API.md

```diff
@@ -29,13 +43,52 @@ export enum CDKContextKey {
     DEPLOYMENT_TYPE = "amplify-backend-type"
 }
 
+// @public
+export class CwdPackageJsonLoader {
+    constructor(fs?: typeof _fs, fsp?: typeof _fsp);
+    loadCwdPackageJson: () => Promise<PackageJson>;
+}
+
+// @public (undocumented)
+export type DeploymentTimes = {
+    synthesisTime?: number;
+    totalTime?: number;
+};
+
 // @public
 export class FilePathExtractor {
     constructor(stackTraceLine: string);
     // (undocumented)
     extract: () => string | undefined;
 }
 
+// @public
+export class LibraryVersionFetcher {
+    // (undocumented)
+    fetch: (absolutePackageJsonPath: string) => string;
+}
+
+// @public (undocumented)
+export type PackageJson = z.infer<typeof packageJsonSchema>;
+
+// @public
+export const packageJsonSchema: z.ZodObject<{
+    name: z.ZodString;
+}, "strip", z.ZodTypeAny, {
+    name: string;
+}, {
+    name: string;
+}>;
+
+// @public
+export class UsageDataEmitter {
+    constructor(libraryVersion: string, sessionUuid?: string, url?: UrlWithStringQuery, accountIdFetcher?: AccountIdFetcher);
+    // (undocumented)
+    emitFailure: (command: string, error: Error) => Promise<void>;
+    // (undocumented)
+    emitSuccess: (command: string, deploymentTimes?: DeploymentTimes, hotswapped?: boolean) => Promise<void>;

```

Comment:
1. if this has `command` in param names then should this type go to `cli-core` instead?
Will we ever emit metrics from constructs or constructs factories?
2. I think that `emitSuccess` should accept `metrics: Record<string, number>` and `dimensions: Record<string, string>` instead of coupling this interface to deployer types.

Reply:
1. Should we move to cli-core? that's where I had it originally but thought it might grow coupled with CLI and there is no reason for other components to emit metrics as well in the future. What do you think?
2. Good idea.

Reply:
Re 1.
If we want to keep it in platform-core we should somehow drop `command` wording. 

I'm fine with moving this to `cli-core` if that's where it belongs now. Even if it somehow gets coupled to CLI then recovery from that would be to extract non-coupled part first and have coupled part + telemetry core.

Reply:
I'm removing the command wording regardless. For now let's keep it in platform-core! Moving to cli-core is much easy than the other way around.